ibm/service/backuprecovery/resource_ibm_backup_recovery.go

```diff
@@ -0,0 +1,5151 @@
+			Required:                   true,
+			AllowedValues:              "kPhysical, kSQL",
+		},
+	)
+
+	resourceValidator := validate.ResourceValidator{ResourceName: "ibm_backup_recovery_recovery", Schema: validateSchema}
+	return &resourceValidator
+}
+
+func resourceIbmBackupRecoveryCreate(context context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	backupRecoveryClient, err := meta.(conns.ClientSession).BackupRecoveryV1()
+	if err != nil {
+		tfErr := flex.DiscriminatedTerraformErrorf(err, err.Error(), "ibm_backup_recovery_recovery", "create", "initialize-client")
+		log.Printf("[DEBUG]\n%s", tfErr.GetDebugMessage())
+		return tfErr.GetDiag()
+	}
+
+	createRecoveryOptions := &backuprecoveryv1.CreateRecoveryOptions{}
+	tenantId := d.Get("x_ibm_tenant_id").(string)
+	createRecoveryOptions.SetXIBMTenantID(tenantId)
+	createRecoveryOptions.SetName(d.Get("name").(string))
+	createRecoveryOptions.SetSnapshotEnvironment(d.Get("snapshot_environment").(string))
+	if _, ok := d.GetOk("physical_params"); ok {
+		physicalParamsModel, err := ResourceIbmBackupRecoveryMapToRecoverPhysicalParams(d.Get("physical_params.0").(map[string]interface{}))
+		if err != nil {
+			return flex.DiscriminatedTerraformErrorf(err, err.Error(), "ibm_backup_recovery_recovery", "create", "parse-physical_params").GetDiag()
+		}
+		createRecoveryOptions.SetPhysicalParams(physicalParamsModel)
+	}
+	if _, ok := d.GetOk("mssql_params"); ok {
+		mssqlParamsModel, err := ResourceIbmBackupRecoveryMapToRecoverSqlParams(d.Get("mssql_params.0").(map[string]interface{}))
+		if err != nil {
+			return flex.DiscriminatedTerraformErrorf(err, err.Error(), "ibm_backup_recovery_recovery", "create", "parse-mssql_params").GetDiag()
+		}
+		createRecoveryOptions.SetMssqlParams(mssqlParamsModel)
+	}
+	if _, ok := d.GetOk("request_initiator_type"); ok {
+		createRecoveryOptions.SetRequestInitiatorType(d.Get("request_initiator_type").(string))
+	}
+
+	recovery, _, err := backupRecoveryClient.CreateRecoveryWithContext(context, createRecoveryOptions)
+	if err != nil {
+		tfErr := flex.TerraformErrorf(err, fmt.Sprintf("CreateRecoveryWithContext failed: %s", err.Error()), "ibm_backup_recovery_recovery", "create")
+		log.Printf("[DEBUG]\n%s", tfErr.GetDebugMessage())
+		return tfErr.GetDiag()
+	}
+
+	recoveryId := fmt.Sprintf("%s::%s", tenantId, *recovery.ID)
+	d.SetId(recoveryId)
+
+	return resourceIbmBackupRecoveryRead(context, d, meta)
+}
+
+func resourceIbmBackupRecoveryRead(context context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	backupRecoveryClient, err := meta.(conns.ClientSession).BackupRecoveryV1()
+	if err != nil {
+		tfErr := flex.DiscriminatedTerraformErrorf(err, err.Error(), "ibm_backup_recovery_recovery", "read", "initialize-client")
+		log.Printf("[DEBUG]\n%s", tfErr.GetDebugMessage())
+		return tfErr.GetDiag()
+	}
+
+	getRecoveryByIdOptions := &backuprecoveryv1.GetRecoveryByIdOptions{}
+	tenantId := d.Get("x_ibm_tenant_id").(string)
```

Comment:
In Read we should avoid doing d.Get it effects the import flow
Already as part of d.SetID we store tenantID why can't we use that?

Reply:
Yes @hkantare , have extracted and used the tenant Id from [d.Id()](https://github.com/IBM-Cloud/terraform-provider-ibm/blob/f6379fc8951e0cade6971fc18bd23cfcbe03ff4c/ibm/service/backuprecovery/resource_ibm_backup_recovery.go#L2496) and used it to set the same variable [tenantId = ParseId(d.Id(), "tenantId")](https://github.com/IBM-Cloud/terraform-provider-ibm/blob/f6379fc8951e0cade6971fc18bd23cfcbe03ff4c/ibm/service/backuprecovery/resource_ibm_backup_recovery.go#L2498)